### PR TITLE
feat(control, signal): control genesis, refactoring

### PR DESCRIPTION
**Pull request by vovacha**
**gamedao-control:**
- add `genesis_build`
- small refactoring related to the `do_create_org`
- `create_org` extrinsic: `gov_asset` and `pay_asset` change type `u8 => CurrencyId`
- error `OrgExists`
- InitialDepost -> MinimumDeposit, make deposit optional

**gamedao-signal:**
- Proposal struct changes: `context_id => org_id + option(campaign_id)`
- Storage `ProposalsByContext* => ProposalsByOrg*.` If needed, later could be also added: `ProposalByType*` or/and `ProposalByCampaign*`
- `create_proposal` refactoring
- migrations

**global:**
- remove randomness, use `hash(nonce)` instead
- remove `gamedao-treasury` pallet
- remove redundant types: `AdminOrigin`, `GameDAOAdminOrigin`, etc.
- tests

**TESTING:**
- build OK
- tests OK
- migration OK

### CODE REVIEW

File: signal/src/lib.rs [hunk 1]
```diff
@@ -677,15 +683,16 @@ pub mod pallet {
 		fn insert_proposal(
 			sender: &T::AccountId,
 			proposal: &Proposal<T::Hash, T::BlockNumber>,
+			org_id: T::Hash,
```
Comment (vasylenko-yevhen) [inline signal/src/lib.rs:686]: Do we even need `org_id` parameter if proposal already contains that?

File: control/src/lib.rs [hunk 2]
```diff
@@ -286,62 +319,37 @@ pub mod pallet {
 			access: AccessModel,
 			fee_model: FeeModel,
 			fee: T::Balance,
-			gov_asset: u8,
-			pay_asset: u8,
+			gov_asset: T::CurrencyId,
+			pay_asset: T::CurrencyId,
 			member_limit: u64,
 			deposit: T::Balance,
```
Comment (vasylenko-yevhen) [inline control/src/lib.rs:325]: What about making this parameter optional and use `T::InitialDeposit::get()` in case it's not provided?

Comment (2075) [reply control/src/lib.rs:325]: maybe also a `T::MinimumDeposit` could be more accurate

**Comment by vovacha** [comment]
We also need migrations for `gamedao-signal,` since storage was changed.
Switching to draft again.

File: signal/src/migration.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+/// V1: Clean up pallet storage
+mod v1 {
+	use super::*;
+	use sp_io::hashing::twox_128;
+
+	pub fn migrate<T: Config>() -> Weight {
+
+		frame_support::storage::unhashed::kill_prefix(
```
Comment (vasylenko-yevhen) [inline signal/src/migration.rs:39]: `kill_prefix` actually returns number of removed storage items so you might use that to calculate weight.

Comment (vovacha) [reply signal/src/migration.rs:39]: `kill_prefix` does exactly one write operation, ex.:
https://github.com/paritytech/substrate/blob/c8653447fc8ef8d95a92fe164c96dffb37919e85/frame/contracts/src/migration.rs#L60-L61
`remove_storage_prefix` calls `frame_support::storage::unhashed::kill_prefix`

**Review by vasylenko-yevhen — APPROVED** [review]

**Pull request merged**